spec/core/v2/ics-004-channel-and-packet-semantics/README.md

```diff
@@ -690,19 +708,25 @@ function recvPacket(
     // Executes Application logic ∀ Payload
     payload=packet.data[0]
     cbs = router.callbacks[payload.destPort]
-    ack,success = cbs.onReceivePacket(packet.channelDestId,packet.channelSourceId,packet.sequence,payload,relayer) // Note that payload includes the version. The application is required to inspect the version to route the data to the proper callback
+    acknowledgement,success = cbs.onReceivePacket(packet.channelDestId,packet.channelSourceId,packet.sequence,payload,relayer) // Note that payload includes the version. The application is required to inspect the version to route the data to the proper callback
+    // construct acknowlegement event by concatenating each app acknowledgment together with a "/" delimiter
+    ackString = ""
+    for i, ack in acknowledgment {
+        ackString = ackString + toHex(ack)
+        if i !== len(acknowledgement) - 1 {
+            ackString = ackString + "/"
+        } 
+    }
     abortTransactionUnless(success)
     if ack != nil {
         // NOTE: Synchronous ack. 
         writeAcknowledgement(packet.channelDestId,packet.sequence,ack)
         // In case of Synchronous ack we emit the event here as we have all the necessary information, while writeAcknowledgement can only retrieve this in case of asynchronous ack. 
-        emitLogEntry("writeAcknowledgement", {
-            sequence: packet.sequence,
+        emitEvents("write_acknowledgement", {
+            sequence: packet.sequence, // value is string in decimal format
```

Comment:
nit/question: Curious about the use of the word "decimal" here. I immediately think of a number with some kind of precision. Would it be simpler to say integer?

Also, what is the reason for using string-formatted numbers here?

Reply:
I mean that the string is base10 as opposed to hexadecimal or something else.

This is assuming the event system is a <string, string> attribute system.

Reply:
Ah, I see. That makes sense.